updater/registry.go

```diff
@@ -171,14 +176,22 @@ func (reg *ResourceRegistry) Purge(keep int) {
 	}
 }
 
-// Reset resets the internal state of the registry, removing all added resources.
-func (reg *ResourceRegistry) Reset() {
+// ResetResources removes all resources from the registry.
+func (reg *ResourceRegistry) ResetResources() {
 	reg.Lock()
 	defer reg.Unlock()
 
 	reg.resources = make(map[string]*Resource)
 }
 
+// ResetIndexes removes all indexes from the registry.
+func (reg *ResourceRegistry) ResetIndexes() {
+	reg.Lock()
+	defer reg.Unlock()
+
+	reg.indexes = make([]Index, 1)
+}
+
```

Comment:
Do you really want to add a index with zero values? I guess you wanted to do `make([]Index, 0, 1)`?